scripts/build-setup.sh

```diff
@@ -44,31 +61,41 @@ do
         -h | --help )
             usage 3 ;;
         riscv-tools | esp-tools)
-            TOOLCHAIN=$1 ;;
-        -ud | --unpinned-deps )
-            USE_PINNED_DEPS=false ;;
+            TOOLCHAIN_TYPE=$1 ;;
         --force | -f | --skip-validate)
             FORCE_FLAG=$1 ;;
-        --skip-conda)
-            SKIP_CONDA=true ;;
-        --skip-toolchain-extra)
-            SKIP_TOOLCHAIN=true ;;
+        --verbose | -v)
+            VERBOSE_FLAG=$1
+            set -x ;;
+        -ud | --use-unpinned-deps )
+            USE_UNPINNED_DEPS=true ;;
+        --skip | -s)
+            shift
+            SKIP_LIST+=(${1}) ;;
         * )
             error "invalid option $1"
             usage 1 ;;
     esac
     shift
 done
 
-if [ "$SKIP_CONDA" = false ]; then
+# check if the arg is found in the SKIP_LIST
+do_skip() {
```

Comment:
This is a bit confusing, since if `do_skip` evaluates to true, the block is run.
Perhaps rename `do_skip` to `run_step`, where `run_step` returns true if the ARG is not in the `SKIP_LIST`.